### PR TITLE
Add data source section

### PR DESCRIPTION
**Pull request by traviolus**

### CODE REVIEW

File: docs/data-provider/data-source/introduction.md [hunk 1]
```diff
@@ -3,3 +3,18 @@ order: 1
 -->
 
 # Introduction
+A data source is the most fundamental unit in BandChain's oracle system. At the simplest level, it is an executable 
+that describes the procedure to retrieve raw data points from a set of primary sources and the fee associated with one 
+data query. These primary sources can either be a traditional API or any other source that returns the desired result.
+
```
Comment (prin-r) [inline docs/data-provider/data-source/introduction.md:16]: should add that the ds is executed off-chain in order to reduce on-chain workload as some tasks cannot be performed on the chain such as heavy computation or anything that takes a long time (network latency, ...).

Comment (traviolus) [reply docs/data-provider/data-source/introduction.md:16]: Done! 🙏🏻

File: docs/data-provider/data-source/implementation.md [hunk 2]
```diff
@@ -3,3 +3,76 @@ order: 2
...
 
 # Implementation
+To better explain the structure and implementation of a data source, let’s look at two possible examples.
+
+### Getting Gold Price Data
+First, let’s look at a data source that queries [GoldPrice.org](https://goldprice.org) for the current price of gold. 
+The script itself is written in Python. Although this data source does not expect any arguments, it can have any number
+of arguments as needed.
+
+```python
```
Comment (prin-r) [inline docs/data-provider/data-source/implementation.md:13]: 1. I think we should make the ds and the os have some relationship with each other.
2. This sample ds takes no input at all, which makes me think that we should add some input to it.
3. I'm currently making an example os with inputs and outputs like this -> {symbol:string,multiplier:u64}/{value:u64}.

Comment (traviolus) [reply docs/data-provider/data-source/implementation.md:13]: Agree with you on this point and I already stated about arguments in the second example. [Total token supply]

Comment (prin-r) [reply docs/data-provider/data-source/implementation.md:13]: Ok then I will use the second ds to continue with the os example

File: docs/data-provider/data-source/introduction.md [hunk 3]
```diff
@@ -3,3 +3,18 @@ order: 1
...
+![](https://i.imgur.com/IaMeqI7.png)
+
+Since BandChain phase 2, we are supporting both permissionless and private APIs, where we introduced an automated and 
+seamless fee collection service for data source providers to generate revenue.
+
+Permissionless API means that they are openly available to anyone who might want to audit the data source, any
```
Comment (cathtunti) [inline docs/data-provider/data-source/introduction.md]: Permissionless APIs are openly available...

Comment (cathtunti) [inline docs/data-provider/data-source/introduction.md]: or

File: docs/data-provider/data-source/deployment.md [hunk 4]
```diff
@@ -3,3 +3,82 @@ order: 3
...
+- the content of the executable to be run by block validators upon receiving a data request for this data source
+- the sender who wish to create the data source
+- the owner of the data source, if specified
+  
+When registering the data source, the message sender can choose whether to specify an owner of the source. 
+If an owner is specified, only the owner can make any changes to the data source once it is registered. 
+They will also be the only party able to collect the accumulated request fees. On the other hand, if an owner is 
+omitted, the data source can no longer be edited after it is registered. 
+
+Note that the sender who creates the data source and the owner of the data source does not need to be the same.
```
Comment (cathtunti) [inline docs/data-provider/data-source/deployment.md]: Note that the sender who creates the data source and the owner of the data source do not need to be the same.

File: docs/data-provider/data-source/deployment.md [hunk 5]
```diff
@@ -3,3 +3,82 @@ order: 3
...
+
+In the case of unowned data sources, it is the data source's configuration on BandChain that cannot be changed.
```
Comment (cathtunti) [inline docs/data-provider/data-source/deployment.md]: didn't we already say this in line 34-35?

**Review by cathtunti — CHANGES_REQUESTED** [review]

**Pull request merged**